libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -2630,9 +2630,21 @@ void ExpressionCompiler::appendExternalFunctionCall(
 	// Check the target contract exists (has code) for non-low-level calls.
 	if (funKind == FunctionType::Kind::External || funKind == FunctionType::Kind::DelegateCall)
 	{
-		m_context << Instruction::DUP1 << Instruction::EXTCODESIZE << Instruction::ISZERO;
-		m_context.appendConditionalRevert(false, "Target contract does not contain code");
-		existenceChecked = true;
+		size_t encodedSize = 0;
+		for (auto const& t: returnTypes)
+			encodedSize += t->decodingType()->calldataHeadSize();
```

Comment:
This is only the head size, right? Seems enough for this check but I'd use a more specific name to make sure no one tries to use it as actual total encoded size by mistake in the future.
```suggestion
		size_t encodedHeadSize = 0;
		for (auto const& t: returnTypes)
			encodedHeadSize += t->decodingType()->calldataHeadSize();
```